h2/src/main/org/h2/tools/CreateCluster.java

```diff
@@ -141,23 +145,40 @@ private void process(String urlSource, String urlTarget,
 
             try {
 
-                // backup
-                Script script = new Script();
-                script.setOut(out);
-                Script.process(connSource, scriptFile, "", "");
-
-                // delete the target database and then restore
+                // Delete the target database first.
                 connTarget = DriverManager.getConnection(
                         urlTarget + ";CLUSTER=''", user, password);
                 statTarget = connTarget.createStatement();
                 statTarget.execute("DROP ALL OBJECTS DELETE FILES");
                 connTarget.close();
 
+                // Backup data from source database in script form.
+                // Start writing to script pipe output end in separate thread.
+                // Variables used from inner class must be final.
+                final PipedReader pipeReader = new PipedReader();
+                final ResultSet rs = statSource.executeQuery("SCRIPT");
+                
+                new Thread(
+                    new Runnable(){
+                        public void run() {
+                            try {
+                                PipedWriter pipeWriter = new PipedWriter(pipeReader);
+                                while (rs.next()) {
+                                    pipeWriter.write(rs.getString(1) + "\n");
+                                }
+                                pipeWriter.close();
```

Comment:
close should be a finally block, otherwise we might leave the reader hanging if something goes wrong